.ci/generate_bear_requirements.py

```diff
@@ -81,6 +82,15 @@ def get_inherited_requirements():
     in_inherited = False
     with open(os.path.join(PROJECT_DIR, 'requirements.txt'), 'r') as file:
         for line in file.read().splitlines():
+            line = line.strip()
+            if not line.startswith('#'):
+                m = re.match('^[A-Za-z._-]+', line)
```

Comment:
Adding more requirements parsing ;-(  Should be using an existing tool for this.
This entire modification to this script wont be necessary when the pyyaml related changes are backed-out.